content/docs/concepts/remote_write_spec_2_0.md

```diff
@@ -0,0 +1,422 @@
+
+### Retries on Partial Writes
+
+Receiver MAY return a 5xx HTTP or 429 HTTP status code on partial write or [partial invalid sample cases](#partial-write) when it expects Sender to retry the whole request. In that case Receiver MUST support idempotency as sender MAY retry with the same request.
+
+### Backward and Forward Compatibility
+
+The protocol follows [semantic versioning 2.0](https://semver.org/): any 2.x compatible Receiver MUST be able to read any 2.x compatible Sender and vice versa. Breaking or backwards incompatible changes will result in a 3.x version of the spec.
+
+The Proto Messages (in Wire Format) themselves are forward / backward compatible, in some respects:
+
+* Removing fields from the proto message requires a major version bump.
+* Adding (optional) fields can be done in a minor version bump.
+
+In other words, this means that future minor versions of 2.x MAY add new optional fields to `io.prometheus.write.v2.Request`, new compressions, Proto Messages and negotiation mechanisms, as long as they are backwards compatible (e.g. optional to both Receiver and Sender).
+
+#### 2.x vs 1.x Compatibility
+
+The 2.x protocol is breaking compatibility with 1.x by introducing a new, mandatory `io.prometheus.write.v2.Request` Proto Message and deprecating the `prometheus.WriteRequest`.
+
+2.x Sender MAY support 1.x Receiver by allowing users to configure what content type Sender should use. 2.x Sender also MAY automatically fall back to different content types, if the Receiver returns 415 HTTP status code.
+
+#### `io.prometheus.write.v2.Request` Proto Schema
+
+<!---
+TODO(bwplotka): Move link to the one on Prometheus main or even buf. 
+-->
+The source of truth is [here](https://github.com/prometheus/prometheus/blob/remote-write-2.0/prompb/io/prometheus/write/v2/types.proto#L32). The `gogo` dependency and options CAN be ignored ([will be removed eventually](https://github.com/prometheus/prometheus/issues/11908)). They are not part of the specification as they don't impact the serialized format.
+
+The simplified version of the new `io.prometheus.write.v2.Request` is presented below.
+
+```
+// Request represents a request to write the given timeseries to a remote destination.
+message Request {
+  // symbols contains a de-duplicated array of string elements used for various
+  // items in a Request message, like labels and metadata items. For the sender's convenience
+  // around empty values for optional fields like unit_ref, symbols array MUST start with
+  // empty string.
+  //
+  // To decode each of the symbolized strings, referenced, by "ref(s)" suffix, you
+  // need to lookup the actual string by index from symbols array. The order of
+  // strings is up to the sender. The receiver should not assume any particular encoding.
+  repeated string symbols = 1;
+  repeated string symbols = 1;
```

Comment:
Is this a typo? Is this field supposed to appear twice?  Thats illegal, surely?

Reply:
copy-pasta, referenced proto does not have it, thanks for finding 🙈